README.md

```diff
@@ -18,9 +18,14 @@ Components included in this package:
 
 This stack is meant for cluster monitoring, so it is pre-configured to collect metrics from all Kubernetes components. In addition to that it delivers a default set of dashboards and alerting rules. Many of the useful dashboards and alerts come from the [kubernetes-mixin project](https://github.com/kubernetes-monitoring/kubernetes-mixin), similar to this project it provides composable jsonnet as a library for users to customize to their needs.
 
+## Warning
+
+`master` branch is under heavy refactoring work. Please use `release-0.7` branch until code refactoring is complete and this information is removed.
```

Comment:
The problem is a lot of our users already use master, and we encouraged them to do that in many releases we said this. I suscept this will cause problems, is there no way to avoid it?

Reply:
Sorry missed the above comments, all for moving fast but might be worth notifying people in a way before this is merged for them to switch from master to latest release? WDYT of cutting a new release and mentioning that in the release notes? Or how long will this state take you think?

Reply:
`release-0.7` is very close to where current `master` is so I don't think a new branch is needed (the diff is in https://github.com/prometheus-operator/kube-prometheus/compare/release-0.7). As for announcing it, it was just done in prometheus community call, this PR mentions it at the top of the readme, and after it is merged I will announce it on slack.

I hope this all will take about a week (depending on how fast reviews will be), there are already 3 pending PRs on top of this one and I have more code on my PC to submit in the next PRs.

Reply:
I'd say additionally to this statement we can announce it in all the channels on kubernetes slack that are related to kube-prometheus. That, plus a commitment to get it done quickly, sounds like a plan to me.

Reply:
Sounds good to me, feel free to ping us on getting the reviews quickly in to not disrupt master for folks, thanks!